src/pyflex/window_selector.py

```diff
@@ -505,6 +543,45 @@ def calculate_ttimes(self):
         self.ttimes = sorted(tts, key=lambda x: x["time"])
         logger.info("Calculated travel times.")
 
+    def reject_on_select_mode(self):
+        """
+        Reject based on select mode. Will reject windows outside of the wave
+        category specified. For example, if self.config.select_mode == "body_only",
+        only body wave windows will be selected.
+        """
+        dist_in_km = geodetics.calcVincentyInverse(
+            self.station.latitude, self.station.longitude, self.event.latitude,
+            self.event.longitude)[0] / 1000.0
+
+        offset = self.event.origin_time - self.observed.stats.starttime
+
+        select_mode = self.config.select_mode
+        if select_mode == "all_waves":
+            min_time = self.ttimes[0]["time"] - self.config.min_period + offset
+            max_time = self.observed.stats.endtime - self.observed.stats.starttime
+        elif select_mode == "body_and_surface":
+            min_time = self.ttimes[0]["time"] - self.config.min_period + offset
+            max_time = dist_in_km / self.config.min_surface_wave_velocity + \
+                    self.config.min_period + offset
+        elif select_mode == "body_wave":
+            min_time = self.ttimes[0]["time"] - self.config.min_period + offset
+            max_time = dist_in_km / self.config.max_surface_wave_velocity + \
+                    3 * self.config.min_period + offset
```

Comment:
> One thing... Do we need to differentiate components? Since for Z and R component, the first arrival is P or Pn. But for T component, it is S or ScS. I think this will bring the code more complexity. I just want to leave a note and we can refine it later if needed...

In theory yes but I would just ignore this. pyflex does not care about components at all and it would not even have to be any of Z,N,E,R, or T. It just checks they are identical for both data and synthetics I believe. The additional complexity is not worth the trade-off.

> Another thing, I just submit an update to FLEXWIN that windows that overlaps with the noise region should be rejected. 

You want at least a bit of overlap to account for acausal filter effects - otherwise you'll never catch the first arrival. If you do that it should be good.

> In pyflex, we have the method --- " reject_on_selection_mode". So if we also also want to put the "reject_on_noise_region". There are two cases when the "reject_on_noise_region" will take effect:
> 1) the rejection_on_selection_mode doesn't take effect(missing station information or something) but windows overlapping with noise region should still be rejected anyway

I agree.

> 2) if people somehow accidently min_time, max_time inside noise region, should pyflex raise an error or leave it out?

Raise an error. I'm a big fan of raise early and force users to fix their mistakes.

A side note: These comments will kind of disappear once you push to this branch and change the line thats commented below. So before you push: can you copy all important comments into the comments section of this issue? It would be nice if they don't get lost!